components/plan-build/bin/hab-plan-build.sh

```diff
@@ -1603,6 +1603,25 @@ _resolve_dependencies() {
   _validate_deps
 }
 
+function do_automatic() {
+  do_default_automatic "$1"
+  return $?
+}
+
+# Loads specified automatic to extend plan DSL.
+function do_default_automatic() {
+  local automatic="${automatic:-$1}"
+
+  if [[ -n "$automatic" ]]; then
```

Comment:
Maybe use `${pkg_automatic:-}` which ends up being an empty string if this variable isn't set. At some point I'd like to update the rest of the code to this style so we can use a `set -u` on the whole program (when you reference an unused variable, the program dies).